code-challenges.test.js

```diff
@@ -14,43 +14,176 @@
+//Function name: accumulatingArray
+//Function takes in an array and returns an array of the accumulating sum. An empty array should return an empty array. 
+// Variables: numbersToAdd1, numbersToAdd2, numbersToAdd3
+// Use map() method to return an array with the same amount of numbers 
+// What I used to better understand the math behind the accumulating sum: 
+//numbersToAdd1
+//at [0], add itself = 2
+//at [1], add itself + [0] = 6
+//at [2], add itself + [1] + [0] = 51 
+//at [3], add itself + [2] + [1] + [0] = 60
+
+//numbersToAdd2
+//at [0], add itself = 0
+//at [1], add itself + [0] = 7
+//at [2], add itself + [1] + [0] = -1
+//at [3], add itself + [2] + [1] + [0] = 11
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("accumulatingArray", () => {
+    // Description of function
+    it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.", () => {
+        // const numbersToAdd1 = [2, 4, 45, 9]
+        // const numbersToAdd2 = [0, 7, -8, 12]
+        // const numbersToAdd3 = []
+        //Function expression: accumulatingArray
+        // const accumulatingArray = (array) => {
+        //     if (array = array.map((element,index) => array.slice(0,index + 1).reduce((a,b) => a + b))) {
+        //      return array
+        //     } else if(array === []) {
+        //         return []
+        //     }
+        // }
+        //Function: accumulatingArray, Variable: numbersToAdd1, expected output
+        expect(accumulatingArray(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        //Function: accumulatingArray, Variable: numbersToAdd2, expected output
+        expect(accumulatingArray(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        //Function: accumulatingArray, Variable: numbersToAdd3, expected output
+        expect(accumulatingArray(numbersToAdd3)).toEqual([])
+      })
+})
+
+// //Fail received: 
+// //accumulatingArray › takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// //ReferenceError: accumulatingArray is not defined
+
 const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+// // Excpected output: [2, 6, 51, 60]
 
 const numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+// // Expected output: [0, 7, -1, 11]
 
 const numbersToAdd3 = []
-// Expected output: []
+// // Expected output: []
 
 
 // b) Create the function that makes the test pass.
+//Function expression: accumulatingArray
+const accumulatingArray = (array) => {
+//Use reduce() method to return the value of the accumulated sum of elements
+    if (array = array.map((element,index) => array.slice(0,index + 1).reduce((a,b) => a + b))) {
+     return array
+//Use conditional statement to return empty array if an empty array is found
+    } else if(array === []) {
+        return []
+    }
+}
```

Comment:
Logic looks great!  It looks like you don't necessarily need the conditional to cover the empty array since map would return an empty array if there was nothing in the array.